job_executor/app.py

```diff
@@ -135,17 +162,14 @@ def main():
             for job in built_jobs + queued_manager_jobs:
                 try:
                     _handle_manager_job(job)
-                except Exception as e:
-                    logger.error(f'{job.job_id} failed')
-                    logger.exception(e)
-                    # TODO: Catch errors in Datastore
-                    #       Only fatal errors should make it here
-                    job_service.update_job_status(
-                        job.job_id, 'failed',
-                        log='Failed due to unexpected error'
+                except Exception as exc:
+                    logger.exception(
+                        f'{job.job_id} failed and could not roll back',
```

Comment:
Jobs will themselves catch all exceptions and handle them by rolling back within their own respective function. If an exception reaches the main loop, that means a rollback has failed (which is catastrophic)

Reply:
I see, that was not easy to understand. I suggest catching a more specific exception on line 165, for example RollbackException so that it is more explicit.

Reply:
I think we might need a catch-all here, but I absolutely see your point! This might be the time for a descriptive comment to save the day?

Reply:
Yes, that would save the day as well 👍